Summarize population by age for population projection

maggiesu0725 opened the pull request.

maggiesu0725 commented:
@lhendey please review

lhendey commented on Prog/reshape ACS data for population projection.do:
```diff
@@ -0,0 +1,568 @@
+
+import excel "L:\Libraries\RegHsg\Raw\ACS 1 year estimate\All_1year.xlsx", sheet("Sheet1") firstrow
+
+label variable HC01_EST_VC01 "totpop"
```
these labels should be written out especially as they mirror the vars created below. you can use the metadata files from the aff download to make this easier.

lhendey commented on Prog/reshape ACS data for population projection.do:
```diff
@@ -0,0 +1,568 @@
+
+import excel "L:\Libraries\RegHsg\Raw\ACS 1 year estimate\ACS_17_1YR_2017.xlsx", sheet("Sheet1") firstrow
+
+label variable HC01_EST_VC01 "totpop"
```
revise labels as above

lhendey commented on Prog/reshape ACS data for population projection.do:
```diff
@@ -0,0 +1,568 @@
+rename HC01_EST_VC01 totpop
+rename HC02_EST_VC01 totmalepop
+rename HC03_EST_VC01 totfemalepop
+
```
add labels for new variables created.

lhendey commented on Prog/reshape ACS data for population projection.do:
```diff
@@ -0,0 +1,568 @@
+import excel "L:\Libraries\RegHsg\Raw\ACS 5 year estimate\All_5year.xlsx", sheet("Sheet1") firstrow
+
+label variable HC01_EST_VC01 "totpop"
+label variable HC02_EST_VC01 "mtotpop"
```
i don't think these are right. hc02_est_vc01 is the percent of the total population according to the metadata.  file:///L:\Libraries\RegHsg\Raw\ACS%201%20year%20estimate\aff_download\ACS_17_1YR_S0101_metadata.csv

lhendey commented on Prog/reshape ACS data for population projection.do:
```diff
@@ -0,0 +1,568 @@
+
+
+gen totunder5= HC01_EST_VC03*HC01_EST_VC01/100
+gen mtotunder5= HC02_EST_VC03*HC01_EST_VC01/100
```
i don't think these are all right. HC02_EST_VC03 is percent of kids under 5 not number of males under 5 according to the metadata.

lhendey commented on Prog/reshape ACS data for population projection.do:
```diff
@@ -0,0 +1,568 @@
+			   
+			   
+			  
+save "D:\DCData\Libraries\RegHsg\Prog\09-16 ACS 5year.dta"
```
save to the data folder not prog.

lhendey commented on Prog/reshape ACS data for population projection.do:
```diff
@@ -0,0 +1,568 @@
+			   
+
+
+save "D:\DCData\Libraries\RegHsg\Prog\08-16 ACS 1year.dta"
```
save to the data folder not prog.

lhendey commented on Prog/reshape ACS data for population projection.do:
```diff
@@ -0,0 +1,568 @@
+
+order _varname v1-v9
+
+save "D:\DCData\Libraries\RegHsg\Prog\ACS_5year_gender.dta"
```
save to data folder.

maggiesu0725 commented:
@lhendey program is updated

lhendey commented on Prog/reshape ACS data for population projection.do:
```diff
@@ -0,0 +1,801 @@
+xpose, clear varname format
+
+order _varname v1-v11
+
```
can you label these vars? for ACS_1year_gender and ACS_5Year_gender?

maggiesu0725 commented:
thanks, label are added @lhendey

lhendey commented:
@maggiesu0725 okay to merge.

maggiesu0725 commented:
new program to calculate race/ethnicity for population projection is ready for review. @lhendey @rpitingolo

rpitingolo commented:
@lhendey thanks. I see your comments.

maggiesu0725 commented:
@lhendey I confirmed with Steven and removed the group quarter restrictions. We will keep the race/ethnicity as is. Since he also wanted to project by jurisdiction I added the Jurisdiction category. Ready for finalizing?

lhendey commented on Prog/Calculate population by age and raceethnicity.sas:
```diff
@@ -129,11 +140,11 @@ run;
 
 
 proc summary data = Race_&year. ;
```
i think you need to put in a proc sort before the proc summary - I'm getting an error that the variables aren't properly sorted.

maggiesu0725 commented:
@lhendey sorry that I need another review. The program now add a new column for Jurisdiction. However, I discovered that 2008-2011 data is missing for Arlington, fairfax, Loudon, Alexandria and Prince Williams. From the meta data it seems like they are available? Could you let me know what possibly is the explanation?

lhendey commented:
@maggiesu0725 - you can't use puma 5100600 for Loudoun in 2008-11.  It contains 3 other counties.  See in your crosswalk - it only makes up 64% of the PUMA.  There is some missing data in several jurisdictions on hispanic/all other for  80+ year olds - check with steven - maybe we need to collapse some of the top age categories?

maggiesu0725 commented:
@lhendey I think you were looking at the 2010 crosswalk. the 2000 crosswalk file is geocorr2014. Loudon takes up 100% of 5100600 in 2000 PUMA.

lhendey commented:
@maggiesu0725 - no i'm looking at the right thing. I'm 100% certain that 5100600 contains more than loudoun county before 2012. look at the output and you can see the drop in population.

maggiesu0725 commented:
It's weird that the crosswalk suggests otherwise. Yes it seems like Loudoun only take up a fraction of 5100600.  Since the other two counties in 5100600 are not in the COG region, we need to take those population out. Is it OK that I use a ratio from https://usa.ipums.org/usa/volii/PUMA_composition_VA.shtml to get the total for Loudoun county? @lhendey

lhendey commented:
@maggiesu0725 that would be fine for total population but not for anything by race/age. I think that its highly likely that Loudoun has a different demographic patterns than the other 3 counties. Ask Steven - but is there a way to just begin doing all the more detailed categories with it in 2012?

maggiesu0725 commented:
Steven suggested in that case we will be only using 2012 onwards data for Loudoun for the projection. @lhendey

lhendey commented:
@maggiesu0725 sounds good.

maggiesu0725 commented:
Are there any changes to this program? I will close this pull request and issue. @lhendey

lhendey commented:
@maggiesu0725 can you remove loudoun 2008-2011 from the output?

maggiesu0725 commented:
data is removed (I replaced data by missing) @lhendey

lhendey commented:
@maggiesu0725 okay - please merge
